src/background/background-init.ts

```diff
@@ -37,21 +36,22 @@ import { getTelemetryClient } from './telemetry/telemetry-client-provider';
 import { TelemetryEventHandler } from './telemetry/telemetry-event-handler';
 import { TelemetryLogger } from './telemetry/telemetry-logger';
 import { TelemetryStateListener } from './telemetry/telemetry-state-listener';
-import { UserStoredDataCleaner } from './user-stored-data-cleaner';
+import { cleanKeysFromStorage } from './user-stored-data-cleaner';
 
 declare var window: Window & InsightsFeatureFlags;
 
 const browserAdapter = new ChromeAdapter();
 const urlValidator = new UrlValidator(browserAdapter);
-const backgroundInitCleaner = new UserStoredDataCleaner(browserAdapter);
 const indexedDBInstance: IndexedDBAPI = new IndexedDBUtil(getIndexedDBStore());
 const indexedDBDataKeysToFetch = [IndexedDBDataKeys.assessmentStore, IndexedDBDataKeys.userConfiguration];
 
-backgroundInitCleaner.cleanUserData(deprecatedStorageDataKeys);
+cleanKeysFromStorage(browserAdapter, deprecatedStorageDataKeys).catch(reason => console.error('error while cleaning user data: ', reason));
 
-// tslint:disable-next-line:no-floating-promises - top-level entry points are intentionally floating promises
-getPersistedData(indexedDBInstance, indexedDBDataKeysToFetch).then((persistedData: PersistedData) => {
-    browserAdapter.getUserData(storageDataKeys, (userData: LocalStorageData) => {
+const persistedDataPromise = getPersistedData(indexedDBInstance, indexedDBDataKeysToFetch);
```

Comment:
do we have a work item to clean up logic in background-init & just use the file for dependency injection 
& not have any logic?

Reply:
also, we don't want to make this sequential?

Reply:
- We don't have work item to clean up background init
- Not sure about the sequential part... it looks like we could but I think that will make the code inside the `then` function more confusing... once you have both `persistedData` and `userData` all the rest of the logic looks simple. Also, using `all` to wait for both promises allow to rearrange initialization steps every way we need/want to.